### PR TITLE
Fix SAR bugs

### PR DESCRIPTION
**Pull request by prodesert22**
- Added tool tip component by @bilalcorbacioglu 
- Fixed png symbol in claim pangochef reward
- Fixed stepper overflow in remove liquidity drawer
- Fixed text warning in remove farm and only show compound button in farms version 3
- Fixed pools cars overflow
- Fixed padding in compound widget
- Added minimium amount to enable compound
- Fixed wallet card overflow
- Close pool details modal only in close button
- Fixed user apr in pools cards/details
- Fixed cFLR not clicked in swap widget
- Fixed compound
- Added locking to pool zero warning
- Fixed remove liquidity input
- Fixed sar protfolio page number
- Replaced deprecated `send` method by `request`

### CODE REVIEW

File: src/hooks/index.tsx [hunk 1]
```diff
@@ -118,3 +120,36 @@ export const useRefetchMinichefSubgraph = () => {
 
   return async () => await queryClient.refetchQueries(['get-minichef-farms-v2', account]);
 };
+
+export function useGetBlockTimestamp(blockNumber?: number) {
+  const latestBlockNumber = useBlockNumber();
+  const _blockNumber = blockNumber ?? latestBlockNumber;
+
+  const { connector } = useWeb3React();
```
Comment (SarjuHansaliya) [inline src/hooks/index.tsx:128]: lets use `useLibrary` here instead of `useWeb3React`.

File: src/hooks/index.tsx [hunk 2]
```diff
@@ -118,3 +120,36 @@ export const useRefetchMinichefSubgraph = () => {
...
+    async function getTimeStamp() {
+      if (!_blockNumber) return;
+
+      const provider =
+        window.ethereum || window.bitkeep?.ethereum || window.xfi?.ethereum || (await connector?.getProvider());
+
+      const result: { timestamp: string } | null = await provider.request({
+        method: 'eth_getBlockByNumber',
+        params: [`0x${_blockNumber.toString(16)}`, false],
+      });
```
Comment (SarjuHansaliya) [inline src/hooks/index.tsx:142]: this method needs to be added to `src/connectors/WalletProviders` to support this method in all chains. Ping me if you have any doubts

File: src/state/ppangoChef/hooks.ts [hunk 3]
```diff
@@ -349,3 +358,64 @@ export function usePangoChefInfos() {
     pairs,
   ]);
 }
+
+export function useUserAPR(stakingInfo?: PangoChefInfo) {
```
Comment (SarjuHansaliya) [inline src/state/ppangoChef/hooks.ts:362]: lets rename this hook to indicate its PangoChef APR something like `useUserPangoChefAPR`

File: src/state/ppangoChef/hooks.ts [hunk 4]
```diff
@@ -349,3 +358,64 @@ export function usePangoChefInfos() {
...
+    const c = blockTime.sub(a);
+    const d = blockTime.sub(b);
+    return BigNumber.from(stakingInfo.stakingApr ?? 0)
+      .mul(c)
+      .div(d)
+      .toString();
+  }, [blockTime, stakingInfo]);
+}
+
+export function useUserRewardRate(stakingInfo: PangoChefInfo) {
```
Comment (SarjuHansaliya) [inline src/state/ppangoChef/hooks.ts:388]: lets rename this hook to indicate its PangoChef reward rate something like `useUserPangoChefRewardRate`

File: src/state/ppangoChef/hooks.ts [hunk 5]
```diff
@@ -349,3 +358,64 @@ export function usePangoChefInfos() {
...
+  const blockTime = useGetBlockTimestamp();
+
+  return useMemo(() => {
+    if (!stakingInfo) return '0';
+
+    const userBalance = stakingInfo.userValueVariables.balance;
+    const userSumOfEntryTimes = stakingInfo.userValueVariables.sumOfEntryTimes;
+
+    const poolBalance = stakingInfo.valueVariables.balance;
+    const poolSumOfEntryTimes = stakingInfo.valueVariables.sumOfEntryTimes;
```
Comment (SarjuHansaliya) [inline src/state/ppangoChef/hooks.ts:372]: make sure to use `?.` everywhere

File: src/state/ppangoChef/hooks.ts [hunk 6]
```diff
@@ -349,3 +358,64 @@ export function usePangoChefInfos() {
...
+
+export function useUserRewardRate(stakingInfo: PangoChefInfo) {
+  const blockTime = useGetBlockTimestamp();
+
+  return useMemo(() => {
+    const userBalance = stakingInfo.userValueVariables.balance;
+    const userSumOfEntryTimes = stakingInfo.userValueVariables.sumOfEntryTimes;
+
+    const poolBalance = stakingInfo.valueVariables.balance;
+    const poolSumOfEntryTimes = stakingInfo.valueVariables.sumOfEntryTimes;
```
Comment (SarjuHansaliya) [inline src/state/ppangoChef/hooks.ts:396]: make sure to use `?.` everywhere

**Review by SarjuHansaliya — CHANGES_REQUESTED** [review]

File: src/components/TextToolTip/index.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  text: string;
+  toolTipText: string;
+  currency?: Currency;
+  color?: ThemeColorsType;
+  cursor?: string;
+}
+
+type Props = TextProps & TypographyProps & SpaceProps & PositionProps;
+
+export default function ToolTipText({ text, toolTipText, currency, ...rest }: Props) {
```
[the remark this answers is not in the document]
Comment (prodesert22) [reply src/components/TextToolTip/index.tsx:18]: How I add this component in my code? just copy it?

[the remark this answers is not in the document]
Comment (SarjuHansaliya) [reply src/components/TextToolTip/index.tsx:18]: yes @prodesert22 for now you can copy paste and use it

**Review by SarjuHansaliya — APPROVED** [review]

**Pull request merged**